### PR TITLE
Diagnostic error events

### PR DESCRIPTION
**Pull request by jacob314**

### CODE REVIEW

**Comment by jacob314** [comment]
For discussion. I've now gone a lot more aggressive about making everything a DiagnosticsNode.
This should provide power to really make the errors rich in editors without breaking error display for console users.

File: packages/flutter/lib/src/widgets/widget_inspector.dart [hunk 1]
```diff
@@ -950,6 +999,10 @@ mixin WidgetInspectorService {
 
     SchedulerBinding.instance.addPersistentFrameCallback(_onFrameStart);
 
+    // We should only actually do this once a service extension enabling it is
+    // turned on but this is a prototype. XXXX.
```
Comment (DanTup) [inline packages/flutter/lib/src/widgets/widget_inspector.dart]: I presume the places here that have `XXX`s are still TODO?

**Review by DanTup — COMMENTED** [review]
I'm not familiar with any of this code (and haven't used it - VS Code only currently displays the errors dumped to stdout of `flutter run) so I'm not sure I can review it well (though on the surface it looks sensible).

I'm happy to test it out in VS Code if you can give me a pointer on how to use it though!

**Comment by InMatrix** [comment]
Is there a high-level overview of the design? Specifically, does the client expect a specific schema from a structured message backed by a list of DiagnosticsNodes?

**Comment by jacob314** [comment]
To be clear this code isn't for review but for discussion.

Here's a small WIP doc describing what I'm doing and why.
https://docs.google.com/document/d/1A5h67yGKgQslw_mBvp2zhGlDqUT6uqkz6V-yak-8gX4

**Pull request closed without merging**